### ISSUE TITLE
Consider adding sphinx-autodoc-typehints to requirements.txt

### ISSUE THREAD

**Issue by stevepiercy** [plone/documentation#1168]
@pauleveritt suggests the use of [`sphinx-autodoc-typehints`](https://github.com/tox-dev/sphinx-autodoc-typehints/) to auto-document docstrings and typehints. In addition to the nice formatting, the return type of one class can be hyperlinked to another in the documentation.

See [tutorial](https://www.jetbrains.com/pycharm/guide/tutorials/sphinx_sites/documentation/) and [updated code samples](https://github.com/talkpython/static-sites-with-sphinx-and-markdown-course/issues/6#issuecomment-1019250374)

**Comment by pauleveritt** [comment]
As a further point, it lessens the work as the docstring doesn't have to repeat type hint information for parameters.

**Comment by stevepiercy** [comment]
Of course, for this extension to be of any use, Plone code needs to use typehints. I have no idea if that has been or is being done.
